homework3/task2/slow_calc.py

```diff
@@ -0,0 +1,41 @@
+'''Here's a not very efficient calculation function that calculates something important:
+
+import time
+import struct
+import random
+import hashlib
+
+def slow_calculate(value):
+    """Some weird voodoo magic calculations"""
+    time.sleep(random.randint(1,3))
+    data = hashlib.md5(str(value).encode()).digest()
+    return sum(struct.unpack('<' + 'B' * len(data), data))
+Calculate total sum of slow_calculate() of all numbers starting from 0 to 500.
+Calculation time should not take more than a minute.
+Use functional capabilities of multiprocessing module.
+You are not allowed to modify slow_calculate function.'''
+
+import hashlib
+import random
+import struct
+import time
+from multiprocessing import Process
+
+
+def slow_calculate(value):
+    """Some weird voodoo magic calculations"""
+    time.sleep(random.randint(1, 3))
+    data = hashlib.md5(str(value).encode()).digest()
+    return sum(struct.unpack("<" + "B" * len(data), data))
+
+
+def slow_sum():
+    jobs = []
+    for i in range(500):
+        p = Process(target=slow_calculate, args=(i,))
+        jobs.append(p)
+        p.start()
```

Comment:
It's good practise to explicitly call `.join()`. Without it, if there would some more calculations in the programm, the resources taken by processes, would not be returned to the main process.